core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp

```diff
@@ -5432,33 +5434,96 @@ lookupswitch_r(JitCompContext *cc, x86::Assembler &a, bh_list *jmp_info_list,
 {
     JmpInfo *node;
     Imm imm;
+    x86::Mem m;
     uint32 i;
-    int32 label_dst;
+    int32 label_dst = 0;
+    char *stream;
+
+    if (opnd->match_pairs_num < 5) {
+        /* For small count of branches, it is better to compare
+           the key with branch value and jump one by one */
+        for (i = 0; i < opnd->match_pairs_num; i++) {
+            imm.setValue(opnd->match_pairs[i].value);
+            a.cmp(regs_i32[reg_no], imm);
+
+            node = (JmpInfo *)jit_malloc(sizeof(JmpInfo));
+            if (!node)
+                GOTO_FAIL;
+
+            node->type = JMP_DST_LABEL_REL;
+            node->label_src = label_src;
+            node->dst_info.label_dst = jit_reg_no(opnd->match_pairs[i].target);
+            node->offset = a.code()->sectionById(0)->buffer().size() + 2;
+            bh_list_insert(jmp_info_list, node);
+
+            imm.setValue(INT32_MAX);
+            a.je(imm);
```

Comment:
in another PR, after `je` to `INT32_MAX`, there is an error checking. Do we need it here?

Reply:
It is because that I don't want to write offset to the invalid place when asmjit error occurs. Normally we check the asm error in jit_codegen_gen_native, after generating machine code for each INSN, in other place, we can skip the check to reduce the footprint.

Reply:
but if the error was raised, https://github.com/bytecodealliance/wasm-micro-runtime/blob/7b72af3d618fe24027854d68309ab990cc0ad62b/core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp#L6513 will be triggered?

Reply:
Yes, it will be triggered, in the beginning of jit_codegen_gen_native, we set the error handler of code buffer:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/7b72af3d618fe24027854d68309ab990cc0ad62b/core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp#L6049-L6051

And the JitErrorHandler is defined here:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/7b72af3d618fe24027854d68309ab990cc0ad62b/core/iwasm/fast-jit/cg/x86-64/jit_codegen_x86_64.cpp#L221-L234